### ISSUE TITLE
Stagger creation or update of Artifactory resources to avoid flooding the API with too many requests

### ISSUE THREAD

**Issue by mehrdgit** [jfrog/terraform-provider-artifactory#103]
Stagger creation or update of Artifactory resources to avoid flooding the API with too many requests that it can not handle.

When there are many artifactory resources that are being created and/or updated the terraform pipeline fails quite frequenlty with the following errors 

│ Error: POST https://<artifactoryurl>:8443/artifactory/api/repositories/<reosource>: 400 [{Status:400 Message:Could not merge and save new descriptor [org.jfrog.common.ExecutionFailed: Last retry failed: Key cdnRedirectRepoConfig is not changeable. Not trying again (code exception)]}]

**Requirements for and issue**
This scenario is easily reproducible.

**Expected behavior**
All resource being created and/or updated without any merge errror.

**Comment by spuder** [comment]
Workaround

```
export TF_CLI_ARGS_apply="-parallelism=1"
```

**Comment by chb0github** [comment]
@spuder - I wouldn't call that a complete work around, as I still see it occur in low parallelism. But, that definitely helps

**Comment by chb0github** [comment]
As discussed in emails, the best approach is to implement client-side retries. It's been something often discussed. When progress is made on this issue I will update it. Right now we have some other things going on.

**Comment by cawilson** [comment]
I've also set the parallelism to 1 when deploying changes, as I kept getting issues with the version of the config out of sync when too many changes were made at once. Since reducing to 1 I haven't seen the issue since. The changes are really quick anyway, so it hasn't been a big deal but yeah, some sort of retry logic may help.

**Comment by chb0github** [comment]
This error is specific to repository creation. I have a fix in place and it seems to be holding - I have tested it thusly:

```hcl
# Required for Terraform 0.13 and up (https://www.terraform.io/upgrade-guides/0-13.html)
terraform {
  required_providers {
    artifactory = {
      source = "registry.terraform.io/jfrog/artifactory"
      version = "2.3.5"
    }
  }
}
variable "supported_repo_types" {
  type = list(string)
  default = [
    "alpine",
    "bower",
    // xray refuses to cargo. They also require a mandatory field we can't currently support
//    "cargo",
    "chef",
    "cocoapods",
    "composer",
    "conan",
    "conda",
    "cran",
    "debian",
    "docker",
    "gems",
    "generic",
    "gitlfs",
    "go",
    "gradle",
    "helm",
    "ivy",
    "maven",
    "npm",
    "nuget",
    "opkg",
    "p2",
    "puppet",
    "pypi",
    // type 'yum' is not to be supported, as this is really of type 'rpm'. When 'yum' is used on create, RT will
    // respond with 'rpm' and thus confuse TF into think there has been a state change.
    "rpm",
    "sbt",
    "vagrant",
    "vcs",
  ]
}
resource "random_id" "randid" {
  byte_length = 16
}


resource "artifactory_local_repository" "local" {
  count = length(var.supported_repo_types)
  key = "${var.supported_repo_types[count.index]}-local"
  package_type = var.supported_repo_types[count.index]
  xray_index = false
  description = "hello ${var.supported_repo_types[count.index]}-local"
}

resource "artifactory_local_repository" "local-rand" {
  count = 100
  key = "foo-${count.index}-local"
  package_type = var.supported_repo_types[random_id.randid.dec % length(var.supported_repo_types)]
  xray_index = true
  description = "hello ${count.index}-local"
}

provider "artifactory" {
  //  supply ARTIFACTORY_USERNAME, _PASSWORD and _URL as env vars
}
resource "artifactory_remote_repository" "npm-remote" {
  key = "npm-remote"
  package_type = "npm"
  url = "https://registry.npmjs.org"
  xray_index = true
}

```

with this command
```bash
terraform apply -parallelism=20
```

producing: 
```
...
artifactory_local_repository.local[9]: Creation complete after 3s [id=docker-local]
artifactory_local_repository.local-rand[8]: Creation complete after 2s [id=foo-8-local]
artifactory_local_repository.local-rand[27]: Creation complete after 13s [id=foo-27-local]
artifactory_local_repository.local[14]: Creation complete after 6s [id=gradle-local]
artifactory_local_repository.local-rand[2]: Creation complete after 3s [id=foo-2-local]
artifactory_local_repository.local-rand[76]: Creation complete after 8s [id=foo-76-local]
artifactory_local_repository.local-rand[60]: Creation complete after 3s [id=foo-60-local]

Apply complete! Resources: 130 added, 0 changed, 0 destroyed.


```
As you can see from the completion time, some took a bit due to complete, but ultimately did.

**Comment by chb0github** [comment]
fixed with release 2.3.5

**Comment by chb0github** [comment]
Please create a new issue with a repeatable HCL - I have tested this with hundreds of repos with high paralellism

**Comment by antonmos** [comment]
if you have it already set up, could you possibly re-run the test? Perhaps, the throttling limits were changed in Artifactory Cloud since you tested it last?

**Comment by antonmos** [comment]
Amazing! thank you!

**Comment by chb0github** [comment]
fixed (again) 2.6.13
